# **Inteligencia Artificial con Aplicaciones en Economía I**

- 👩‍🏫 **Profesora:** [Lina María Castro](https://www.linkedin.com/in/lina-maria-castro)  
- 📧 **Email:** [lmcastroco@gmail.com](mailto:lmcastroco@gmail.com)  
- 🎓 **Universidad:** Universidad Externado de Colombia - Facultad de Economía

# **Taller Limpieza y transformación de datos**🧹

**IMPORTANTE**: Guarda una copia de este notebook en tu Google Drive o computador.

**Taller en grupos de 3 o parejas.**

**Nombres estudiantes:**

- Camilo Rendón C.
- Nicolás Cubillos R.
-

**Forma de entrega:**

- Nombrar el archivo de la siguiente forma:“Taller_Limpieza_nombres_y_appellidos.ipynb”.
- Suba el Jupyter Notebook a su cuenta en Github y envíe el link en el siguiente Forms: https://forms.office.com/r/6GZ89uiLNC.

**IMPORTANTE:** No se recibirán talleres en Google Colab, el notebook debe estar subido en Github.

**Plazo de entrega:**

19 de marzo de 2026, máximo a las 11:59 p.m. Tenga en cuenta que luego de esa hora el formulario en forms se cierra. El Jupupyter Notebook también debe quedar subido en Github antes de esa hora.

**Instrucciones Generales:**

Completa el código en las celdas marcadas con `### TU CÓDIGO AQUÍ ###`. Puedes añadir más celdas si lo requieres.

Utilice la base de la GEIH de julio a diciembre de 2025, la cual se encuentra en https://drive.google.com/drive/folders/1Th4TWdsOahAAvBUKxixFhaELC_KPsl3y?usp=sharing. Tenga en cuenta sep="," y encoding='latin1'. Las variables 'DPTO' y 'RAMA4D_R4' deben ser cargadas como string.

# Ejercicio 1: Importar librerías y cargar datos

1.	Importe todas las librerías que vaya a utilizar.

In [ ]:
import os
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

2.	Mejore la visualización de los dataframes como se ha visto en clase.

In [ ]:
# Que muestre todas las columnas
pd.options.display.max_columns = None
# En los dataframes, mostrar los float con dos decimales
pd.options.display.float_format = '{:,.2f}'.format


3.	Conéctese a su Google Drive y establezca la ruta del dataset.

In [ ]:
from google.colab import drive, files
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
path = '/content/drive/MyDrive/Bases'

In [ ]:
os.chdir(path)

4.	Cargue los datos en el notebook. Llame al dataframe 'df_geih'

In [ ]:
df_geih = pd.read_csv('/content/drive/MyDrive/Bases/GEIH_2025_Jul_Dic_Consolidado.csv', encoding='latin1')

5. Obtenga las dimensiones de la base

In [ ]:
df_geih.shape

(409127, 314)

# Ejercicio 2:  Seleccionar y renombrar columnas

La GEIH contiene cientos de variables. Para nuestro análisis del mercado laboral, nos enfocaremos en un subconjunto.

Adicionalmente, los nombres de las columnas en la GEIH son códigos. Para hacer el análisis más intuitivo, vamos a renombrarlas.

1. Crea un nuevo DataFrame llamado `df_empleo` que contenga únicamente las columnas que están en la lista `columnas_interes`.

In [ ]:
columnas_interes = ['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'MES', 'CLASE', 'DPTO', 'FEX_C18', 'PT', 'P3271', 'P6040', 'P6080', 'P3043', 'OCI',
                    'P6430', 'RAMA4D_R4', 'INGLABO', 'DESCRIPCION_CIIU', 'NOMBRE_DPTO']

In [ ]:
df_empleo = df_geih[columnas_interes]
df_empleo


,DIRECTORIO,SECUENCIA_P,ORDEN,MES,CLASE,DPTO,FEX_C18,PT,P3271,P6040,P6080,P3043,OCI,P6430,RAMA4D_R4,INGLABO,DESCRIPCION_CIIU,NOMBRE_DPTO
0,8283143,1,1,7,2,8,27.75,1,1,72,6,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,2,8,27.75,1,2,34,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
2,8283145,1,1,7,2,8,59.11,1,1,45,6,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,2,8,59.11,1,2,50,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
4,8283145,1,3,7,2,8,59.11,1,1,19,6,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,2,73,378.31,1,2,61,6,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,1,54,534.55,1,1,58,6,9.00,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,1,54,534.55,1,2,51,6,9.00,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,1,54,534.55,1,2,31,6,7.00,NaN,NaN,NaN,NaN,NaN,NORTE DE SANTANDER


2. Renombra las columnas del DataFrame `df_empleo` utilizando el diccionario `mapa_nombres`.

In [ ]:
mapa_nombres = {
'DIRECTORIO': 'id_vivienda',
'SECUENCIA_P': 'id_hogar',
'ORDEN': 'id_persona',
'MES': 'mes',
'CLASE': 'clase',
'DPTO': 'departamento',
'FEX_C18': 'factor_expansion',
'PT': 'poblacion_total',
'P3271': 'sexo',
'P6040': 'edad',
'P6080': 'grupo_etnico',
'P3043': 'nivel_educativo',
'OCI': 'ocupados',
'P6430': 'categoria_ocupacional',
'RAMA4D_R4': 'cod_ciiu',
'INGLABO': 'ingreso_laboral',
'DESCRIPCION_CIIU': 'desc_ciiu',
'NOMBRE_DPTO': 'nombre_dpto'}

In [ ]:
df_empleo.rename(columns=mapa_nombres, inplace=True)
df_empleo


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,2,8,27.75,1,1,72,6,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,2,8,27.75,1,2,34,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
2,8283145,1,1,7,2,8,59.11,1,1,45,6,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,2,8,59.11,1,2,50,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
4,8283145,1,3,7,2,8,59.11,1,1,19,6,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,2,73,378.31,1,2,61,6,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,1,54,534.55,1,1,58,6,9.00,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,1,54,534.55,1,2,51,6,9.00,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,1,54,534.55,1,2,31,6,7.00,NaN,NaN,NaN,NaN,NaN,NORTE DE SANTANDER


3. Obtenga las dimensiones de la base

In [ ]:
df_empleo.shape


(409127, 18)

# Ejercicio 3: Revisión de duplicados

Un paso crucial en la limpieza es asegurar que no haya observaciones duplicadas que puedan sesgar nuestro análisis.

1. Verifica si hay filas duplicadas en `df_empleo`.

In [ ]:
df_empleo.duplicated()


,0
0,False
1,False
2,False
3,False
4,False
...,...
409122,False
409123,False
409124,False
409125,False


2. Cuenta cuántas filas duplicadas existen.

In [ ]:
df_empleo.duplicated().sum()


np.int64(0)

3. Elimina las filas duplicadas, si las hay.

In [ ]:
df_empleo.drop_duplicates(inplace=True)
df_empleo.shape

(409127, 18)

4. Verifica si hay filas duplicadas específicamente en las columnas 'id_vivienda', 'id_hogar', 'id_persona'.

In [ ]:
df_empleo.duplicated(subset=['id_vivienda', 'id_hogar', 'id_persona'])


,0
0,False
1,False
2,False
3,False
4,False
...,...
409122,False
409123,False
409124,False
409125,False


# Ejercicio 4: Valores faltantes

Los datos del mundo real raramente están completos. Aquí, trataremos los valores faltantes en la columna de ingresos.


1. Calcula el número de valores nulos (`NaN`) en la columna `ingreso_laboral` **antes** de cualquier modificación.

In [ ]:
df_empleo['ingreso_laboral'].isna().sum()


np.int64(231144)

2. En la GEIH, un ingreso faltante a menudo significa que la persona no reportó ingresos o no aplica (ej. no trabaja). Una estrategia común es imputar estos valores con `0`. Rellena los valores `NaN` en la columna `ingreso_laboral` con `0`.

In [ ]:
df_empleo['ingreso_laboral'].fillna(0, inplace=True)
df_empleo


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,2,8,27.75,1,1,72,6,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,2,8,27.75,1,2,34,6,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
2,8283145,1,1,7,2,8,59.11,1,1,45,6,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,2,8,59.11,1,2,50,6,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
4,8283145,1,3,7,2,8,59.11,1,1,19,6,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,2,73,378.31,1,2,61,6,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,1,54,534.55,1,1,58,6,9.00,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,1,54,534.55,1,2,51,6,9.00,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,1,54,534.55,1,2,31,6,7.00,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER


# Ejercicio 5: Decodificar variables



Las variables categóricas a menudo necesitan ser "traducidas" para hacerlas más entendibles para el análisis.

1. Usa los siguientes diccionarios para reemplazar los códigos numéricos por etiquetas de texto en las siguientes columnas: 'clase', 'sexo', 'grupo_etnico', 'nivel_educativo' y 'categoria_ocupacional'. Muestra los conteos de las nuevas categorías usando `.value_counts()` para verificar el cambio en cada una de las columnas.

In [ ]:
dic_clase = {1: 'Cabeceras', 2: 'Centros poblados y rural disperso'}
dic_sexo = {1: 'Hombre', 2: 'Mujer'}
dic_etnico = {1: 'Indígena', 2: 'Gitano (Rom)', 3: 'Raizal del archipiélago', 4: 'Palenquero',
              5: 'Negro, mulato (afrodescendiente)', 6: 'Ninguno de los anteriores'}
dic_nivel_educativo = {1: 'Ninguno', 2: 'Educación media', 3: 'Educación media',
                       4: 'Educación media', 5: 'Técnica profesional', 6: 'Tecnológica',
                       7: 'Universitaria', 8: 'Especialización', 9: 'Maestría',
                       10: 'Doctorado', 99: 'No sabe, no informa'}
dic_cat_ocu = {1: 'Asalariado', 2: 'Asalariado', 3: 'Asalariado',
               8: 'Asalariado', 4: 'Independiente', 5: 'Independiente',
               9: 'Independiente', 6: 'Trabajador sin remuneración',
               7: 'Trabajador sin remuneración'}

In [ ]:
df_empleo['clase'].replace(dic_clase, inplace=True)
df_empleo

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,Centros poblados y rural disperso,8,27.75,1,1,72,6,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,Centros poblados y rural disperso,8,27.75,1,2,34,6,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
2,8283145,1,1,7,Centros poblados y rural disperso,8,59.11,1,1,45,6,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,Centros poblados y rural disperso,8,59.11,1,2,50,6,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
4,8283145,1,3,7,Centros poblados y rural disperso,8,59.11,1,1,19,6,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,Centros poblados y rural disperso,73,378.31,1,2,61,6,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,Cabeceras,54,534.55,1,1,58,6,9.00,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,Cabeceras,54,534.55,1,2,51,6,9.00,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,Cabeceras,54,534.55,1,2,31,6,7.00,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER


In [ ]:
df_empleo['sexo'].replace(dic_sexo, inplace=True)
df_empleo

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,Centros poblados y rural disperso,8,27.75,1,Hombre,72,6,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,Centros poblados y rural disperso,8,27.75,1,Mujer,34,6,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
2,8283145,1,1,7,Centros poblados y rural disperso,8,59.11,1,Hombre,45,6,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,Centros poblados y rural disperso,8,59.11,1,Mujer,50,6,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
4,8283145,1,3,7,Centros poblados y rural disperso,8,59.11,1,Hombre,19,6,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,Centros poblados y rural disperso,73,378.31,1,Mujer,61,6,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,Cabeceras,54,534.55,1,Hombre,58,6,9.00,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,Cabeceras,54,534.55,1,Mujer,51,6,9.00,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,Cabeceras,54,534.55,1,Mujer,31,6,7.00,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER


In [ ]:
df_empleo['grupo_etnico'].replace(dic_etnico, inplace=True)
df_empleo


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,Centros poblados y rural disperso,8,27.75,1,Hombre,72,Ninguno de los anteriores,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,Centros poblados y rural disperso,8,27.75,1,Mujer,34,Ninguno de los anteriores,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
2,8283145,1,1,7,Centros poblados y rural disperso,8,59.11,1,Hombre,45,Ninguno de los anteriores,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,Centros poblados y rural disperso,8,59.11,1,Mujer,50,Ninguno de los anteriores,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
4,8283145,1,3,7,Centros poblados y rural disperso,8,59.11,1,Hombre,19,Ninguno de los anteriores,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,Centros poblados y rural disperso,73,378.31,1,Mujer,61,Ninguno de los anteriores,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,Cabeceras,54,534.55,1,Hombre,58,Ninguno de los anteriores,9.00,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,Cabeceras,54,534.55,1,Mujer,51,Ninguno de los anteriores,9.00,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,Cabeceras,54,534.55,1,Mujer,31,Ninguno de los anteriores,7.00,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER


In [ ]:
df_empleo['nivel_educativo'].replace(dic_nivel_educativo, inplace=True)
df_empleo


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,Centros poblados y rural disperso,8,27.75,1,Hombre,72,Ninguno de los anteriores,NaN,1.00,7.00,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,Centros poblados y rural disperso,8,27.75,1,Mujer,34,Ninguno de los anteriores,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
2,8283145,1,1,7,Centros poblados y rural disperso,8,59.11,1,Hombre,45,Ninguno de los anteriores,NaN,1.00,1.00,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,Centros poblados y rural disperso,8,59.11,1,Mujer,50,Ninguno de los anteriores,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
4,8283145,1,3,7,Centros poblados y rural disperso,8,59.11,1,Hombre,19,Ninguno de los anteriores,NaN,1.00,7.00,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,Centros poblados y rural disperso,73,378.31,1,Mujer,61,Ninguno de los anteriores,NaN,1.00,3.00,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,Cabeceras,54,534.55,1,Hombre,58,Ninguno de los anteriores,Maestría,1.00,2.00,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,Cabeceras,54,534.55,1,Mujer,51,Ninguno de los anteriores,Maestría,1.00,2.00,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,Cabeceras,54,534.55,1,Mujer,31,Ninguno de los anteriores,Universitaria,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER


In [ ]:
df_empleo['categoria_ocupacional'].replace(dic_cat_ocu, inplace=True)
df_empleo


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8283143,1,1,7,Centros poblados y rural disperso,8,27.75,1,Hombre,72,Ninguno de los anteriores,NaN,1.00,Trabajador sin remuneración,143.00,"790,000.00",Cría de ovejas y cabras,ATLÁNTICO
1,8283143,1,2,7,Centros poblados y rural disperso,8,27.75,1,Mujer,34,Ninguno de los anteriores,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
2,8283145,1,1,7,Centros poblados y rural disperso,8,59.11,1,Hombre,45,Ninguno de los anteriores,NaN,1.00,Asalariado,141.00,"1,100,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
3,8283145,1,2,7,Centros poblados y rural disperso,8,59.11,1,Mujer,50,Ninguno de los anteriores,NaN,NaN,NaN,NaN,0.00,NaN,ATLÁNTICO
4,8283145,1,3,7,Centros poblados y rural disperso,8,59.11,1,Hombre,19,Ninguno de los anteriores,NaN,1.00,Trabajador sin remuneración,141.00,"820,000.00",Cría de ganado bovino y bufalino,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,Centros poblados y rural disperso,73,378.31,1,Mujer,61,Ninguno de los anteriores,NaN,1.00,Asalariado,"9,700.00","1,400,000.00",Actividades de los hogares individuales como e...,TOLIMA
409123,8473908,1,1,12,Cabeceras,54,534.55,1,Hombre,58,Ninguno de los anteriores,Maestría,1.00,Asalariado,"8,530.00","3,000,000.00",Establecimientos que combinan diferentes nivel...,NORTE DE SANTANDER
409124,8473908,1,2,12,Cabeceras,54,534.55,1,Mujer,51,Ninguno de los anteriores,Maestría,1.00,Asalariado,"8,412.00","7,956,000.00",Actividades ejecutivas de la administración pú...,NORTE DE SANTANDER
409125,8473908,1,3,12,Cabeceras,54,534.55,1,Mujer,31,Ninguno de los anteriores,Universitaria,NaN,NaN,NaN,0.00,NaN,NORTE DE SANTANDER


# Ejercicio 6: Reorganizar columnas

1. Reorganiza las columnas con el siguiente orden: 'id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento', 'nombre_dpto',     'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico', 'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu', 'ingreso_laboral'

In [ ]:
df_empleo = df_empleo[['id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento', 'nombre_dpto', 'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico', 'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu', 'ingreso_laboral']]
df_empleo

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
0,8283143,1,1,7,2,8,ATLÁNTICO,27.75,1,1,72,6,NaN,1.00,7.00,143.00,Cría de ovejas y cabras,"790,000.00"
1,8283143,1,2,7,2,8,ATLÁNTICO,27.75,1,2,34,6,NaN,NaN,NaN,NaN,NaN,NaN
2,8283145,1,1,7,2,8,ATLÁNTICO,59.11,1,1,45,6,NaN,1.00,1.00,141.00,Cría de ganado bovino y bufalino,"1,100,000.00"
3,8283145,1,2,7,2,8,ATLÁNTICO,59.11,1,2,50,6,NaN,NaN,NaN,NaN,NaN,NaN
4,8283145,1,3,7,2,8,ATLÁNTICO,59.11,1,1,19,6,NaN,1.00,7.00,141.00,Cría de ganado bovino y bufalino,"820,000.00"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409122,8473907,1,1,12,2,73,TOLIMA,378.31,1,2,61,6,NaN,1.00,3.00,"9,700.00",Actividades de los hogares individuales como e...,"1,400,000.00"
409123,8473908,1,1,12,1,54,NORTE DE SANTANDER,534.55,1,1,58,6,9.00,1.00,2.00,"8,530.00",Establecimientos que combinan diferentes nivel...,"3,000,000.00"
409124,8473908,1,2,12,1,54,NORTE DE SANTANDER,534.55,1,2,51,6,9.00,1.00,2.00,"8,412.00",Actividades ejecutivas de la administración pú...,"7,956,000.00"
409125,8473908,1,3,12,1,54,NORTE DE SANTANDER,534.55,1,2,31,6,7.00,NaN,NaN,NaN,NaN,NaN


# Ejercicio 7: Cambiar tipo de dato


Para realizar cálculos y análisis correctamente, las columnas deben tener el tipo de dato adecuado (numérico, categórico, etc.).

1.  Muestra en pantalla los tipos de datos actuales de `df_empleo`.

In [ ]:
df_empleo.dtypes


,0
id_vivienda,int64
id_hogar,int64
id_persona,int64
mes,int64
clase,object
departamento,int64
nombre_dpto,object
factor_expansion,float64
poblacion_total,int64
sexo,object


2.  Las siguientes columnas representan categorías: 'clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu'. Conviértelas al tipo `category` para optimizar el uso de memoria y facilitar análisis futuros.

In [ ]:

df_empleo[['clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu']] = df_empleo [['clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu']].astype('category')
df_empleo.dtypes

,0
id_vivienda,int64
id_hogar,int64
id_persona,int64
mes,int64
clase,category
departamento,category
nombre_dpto,category
factor_expansion,float64
poblacion_total,int64
sexo,category


# Ejercicio 8: Ordenar el dataframe (5%)

1. Ordena el dataframe por la columna "edad" de mayor a menor. Observa la columna 'edad' y escribe en una celda de markdown cuál fue la edad mínima y la edad máxima de las personas que respondieron la encuesta de la GEIH.

In [ ]:
df_empleo.sort_values(by='edad', ascending=False)


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
256896,8395695,1,1,10,Cabeceras,18,CAQUETÁ,40.61,1,Mujer,108,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
5135,8286476,1,5,7,Cabeceras,27,CHOCÓ,31.65,1,Mujer,108,"Negro, mulato (afrodescendiente)",NaN,NaN,NaN,NaN,NaN,0.00
32904,8299104,1,3,7,Centros poblados y rural disperso,13,BOLÍVAR,128.58,1,Mujer,106,"Negro, mulato (afrodescendiente)",NaN,NaN,NaN,NaN,NaN,0.00
256662,8395585,1,3,10,Cabeceras,17,CALDAS,66.42,1,Mujer,106,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
396633,8468208,1,4,12,Centros poblados y rural disperso,25,CUNDINAMARCA,"1,231.96",1,Mujer,106,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
245532,8391286,1,5,10,Cabeceras,63,QUINDÍO,151.42,1,Mujer,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
145507,8348957,1,4,9,Cabeceras,50,META,241.71,1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
185317,8365558,1,3,9,Cabeceras,85,CASANARE,299.88,1,Mujer,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00
366733,8446239,1,6,12,Cabeceras,8,ATLÁNTICO,"1,371.41",1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,0.00


In [ ]:
edad_min = df_empleo["edad"].min()
edad_max = df_empleo["edad"].max()

edad_min, edad_max

(0, 108)

La edad mínima de las personas encuestadas es **0 años** y la edad máxima es **108 años**.


# Ejercicio 9: Groupby

Agrupar los datos por categorías, nos permite realizar análisis más profundos y obtener insights valiosos, que a simple vista no vemos en la base de datos.

1. Crea un nuevo dataframe llamado "df_sexo" donde se agrupe por 'sexo' y se calcule la suma de las columnas 'poblacion_total' y 'ocupados'. Crea una nueva columna que se llame 'porcentaje_ocupados' donde dividas las columna 'ocupados' sobre 'poblacion_total'. Al final usa reset_index() para convertir las categorías de sexo en una columna.

In [ ]:
df_sexo = df_empleo.groupby('sexo')[['poblacion_total', 'ocupados']].sum().reset_index()
df_sexo['porcentaje_ocupados'] = df_sexo['ocupados'] / df_sexo['poblacion_total']
df_sexo


,sexo,poblacion_total,ocupados,porcentaje_ocupados
0,Hombre,190530,"100,790.00",0.53
1,Mujer,218597,"81,063.00",0.37


2. Crea un nuevo dataframe llamado "df_educacion" donde se agrupe por 'nivel_educativo' y se calcule la suma de las columnas 'poblacion_total' y 'ocupados'. Crea una nueva columna que se llame 'porcentaje_ocupados' donde dividas las columna 'ocupados' sobre 'poblacion_total'. Al final usa reset_index() para convertir las categorías de nivel educativo en una columna.

In [ ]:
df_educacion = df_empleo.groupby('nivel_educativo')[['poblacion_total', 'ocupados']].sum().reset_index()
df_educacion['porcentaje_ocupados'] = df_educacion['ocupados'] / df_educacion['poblacion_total']
df_educacion


,nivel_educativo,poblacion_total,ocupados,porcentaje_ocupados
0,Doctorado,339,294.00,0.87
1,Educación media,116635,"66,945.00",0.57
2,Especialización,7927,"6,269.00",0.79
3,Maestría,3154,"2,701.00",0.86
4,Ninguno,12963,"2,539.00",0.20
5,"No sabe, no informa",1,1.00,1.00
6,Tecnológica,10640,"7,928.00",0.75
7,Técnica profesional,24503,"16,989.00",0.69
8,Universitaria,32517,"23,461.00",0.72


3. En una celda de Markdown escribe qué conclusiones puedes sacar de la información obtenida en df_sexo y df_educacion.

Los datos revelan una brecha de género pronunciada y una fuerte dependencia de la formación académica para acceder al empleo. Mientras que los hombres presentan una tasa de ocupación del 53%, las mujeres, a pesar de ser más numerosas en la población total, solo alcanzan un 37%, lo que evidencia barreras estructurales para la equidad laboral. Paralelamente, la educación actúa como el principal motor de empleabilidad: los niveles de postgrado (Doctorado y Maestría) garantizan tasas de ocupación superiores al 85%, mientras que las personas sin formación académica apenas logran un 20% de inserción. Esta tendencia subraya que, aunque la Educación Media aporta el mayor volumen de trabajadores al mercado, la especialización profesional es el factor determinante para asegurar un puesto de trabajo.

Considerando estos datos, el grupo de Educación Media es el que más personas aporta al total de ocupados (66,945), pero proporcionalmente tiene menos éxito que un profesional universitario. Por otro lado, la diferencia de 16 puntos porcentuales entre hombres y mujeres sugiere que, a pesar de haber más mujeres en la base poblacional, el mercado laboral sigue favoreciendo la contratación masculina.

# Ejercicio 10: Filtrar datos

1. A partir de df_empleo, crea el DataFrame `df_ing` con observaciones donde `ingreso_laboral` no sea nulo y además sea mayor estrictamente al salario mínimo para 2026 (con subsidio de transporte).

In [ ]:
df_ing = df_empleo[df_empleo['ingreso_laboral'].notnull() & (df_empleo['ingreso_laboral'] > 2000000)]
df_ing



,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
6,8283147,1,1,7,Centros poblados y rural disperso,8,ATLÁNTICO,57.99,1,Hombre,36,Ninguno de los anteriores,NaN,1.00,Asalariado,"5,022.00",Transporte fluvial de carga,"2,623,500.00"
41,8283163,1,1,7,Centros poblados y rural disperso,8,ATLÁNTICO,496.47,1,Hombre,39,Ninguno de los anteriores,Tecnológica,1.00,Asalariado,"1,103.00","Producción de malta, elaboración de cervezas y...","3,000,000.00"
111,8283213,1,2,7,Cabeceras,66,RISARALDA,360.58,1,Hombre,24,Ninguno de los anteriores,Universitaria,1.00,Asalariado,"8,610.00","Actividades de hospitales y clínicas, con inte...","2,700,000.00"
204,8283275,1,2,7,Cabeceras,66,RISARALDA,208.52,1,Mujer,35,Ninguno de los anteriores,Técnica profesional,1.00,Asalariado,"4,631.00",Comercio al por mayor de productos alimenticios,"2,200,000.00"
205,8283277,1,1,7,Centros poblados y rural disperso,66,RISARALDA,537.44,1,Hombre,69,Ninguno de los anteriores,Educación media,1.00,Asalariado,"5,611.00",Expendio a la mesa de comidas preparadas,"2,450,000.00"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409096,8473895,1,1,12,Cabeceras,11,"BOGOTÁ, D.C.","3,528.95",1,Hombre,60,Ninguno de los anteriores,Educación media,1.00,Asalariado,"4,921.00",Transporte de pasajeros,"2,700,000.00"
409105,8473898,1,1,12,Cabeceras,17,CALDAS,275.62,1,Mujer,30,Ninguno de los anteriores,Universitaria,1.00,Asalariado,"1,410.00","Confección de prendas de vestir, excepto prend...","3,800,000.00"
409120,8473904,1,1,12,Cabeceras,91,AMAZONAS,64.83,1,Mujer,55,Ninguno de los anteriores,Maestría,1.00,Asalariado,"8,551.00",Formación para el trabajo,"5,000,000.00"
409123,8473908,1,1,12,Cabeceras,54,NORTE DE SANTANDER,534.55,1,Hombre,58,Ninguno de los anteriores,Maestría,1.00,Asalariado,"8,530.00",Establecimientos que combinan diferentes nivel...,"3,000,000.00"


2. ¿Cuántas personas tienen un salario estrictamente mayor al salario mínimo?

In [ ]:
salario_minimo = 2000000

conteo = (df_ing['ingreso_laboral'] > salario_minimo).sum()

print(conteo)


38093


3. Crea el dataframe df_mujeres donde se encuentren solamente mujeres que están ubicadas en el departamento del Tolima, que son asalariadas y cuyo nivel educativo es educación media o superior.

In [ ]:
niveles_validos = [
    'Educación media',
    'Universitaria',
    'Técnica profesional',
    'Tecnológica',
    'Especialización',
    'Maestría',
    'Doctorado'
]

df_mujeres = df_empleo[
    (df_empleo['sexo'] == 'Mujer') &
    (df_empleo['nombre_dpto'] == 'TOLIMA') &
    (df_empleo['categoria_ocupacional'] == 'Asalariado') &
    (df_empleo['nivel_educativo'].isin(niveles_validos))
]
df_mujeres

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
9253,8289007,1,3,7,Cabeceras,73,TOLIMA,240.08,1,Mujer,25,Ninguno de los anteriores,Universitaria,1.00,Asalariado,"4,321.00",Instalaciones eléctricas,"3,500,000.00"
9269,8289014,1,3,7,Cabeceras,73,TOLIMA,130.81,1,Mujer,28,Ninguno de los anteriores,Universitaria,1.00,Asalariado,"8,530.00",Establecimientos que combinan diferentes nivel...,"1,600,000.00"
9272,8289015,1,2,7,Cabeceras,73,TOLIMA,221.38,1,Mujer,27,Ninguno de los anteriores,Técnica profesional,1.00,Asalariado,"4,773.00",Comercio al por menor de productos farmacéutic...,"1,423,500.00"
9286,8289027,1,1,7,Cabeceras,73,TOLIMA,240.63,1,Mujer,58,Ninguno de los anteriores,Técnica profesional,1.00,Asalariado,"3,600.00","Captación, tratamiento y distribución de agua","2,000,000.00"
9287,8289029,1,1,7,Cabeceras,73,TOLIMA,285.72,1,Mujer,41,Ninguno de los anteriores,Educación media,1.00,Asalariado,"5,611.00",Expendio a la mesa de comidas preparadas,"1,673,000.00"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
409007,8473866,1,1,12,Cabeceras,73,TOLIMA,309.45,1,Mujer,37,Ninguno de los anteriores,Técnica profesional,1.00,Asalariado,"8,530.00",Establecimientos que combinan diferentes nivel...,"1,900,000.00"
409012,8473867,1,2,12,Cabeceras,73,TOLIMA,279.33,1,Mujer,63,Ninguno de los anteriores,Universitaria,1.00,Asalariado,"8,530.00",Establecimientos que combinan diferentes nivel...,"5,500,000.00"
409014,8473868,1,2,12,Cabeceras,73,TOLIMA,293.95,1,Mujer,65,Ninguno de los anteriores,Universitaria,1.00,Asalariado,"8,413.00",Regulación de las actividades de organismos qu...,"4,000,000.00"
409018,8473869,1,2,12,Cabeceras,73,TOLIMA,621.33,1,Mujer,49,Ninguno de los anteriores,Educación media,1.00,Asalariado,"8,121.00",Limpieza general interior de edificios,"1,600,000.00"


4. ¿Cuáles son las dimensiones de df_mujeres?

In [ ]:
df_mujeres.shape


(1137, 18)